### PR TITLE
Ab#82001 create a blob storage service

### PR DESCRIPTION
**Pull request by tc-chris**

### CODE REVIEW

File: DocumentService/Controllers/DocumentsController.cs [hunk 1]
```diff
@@ -12,10 +14,23 @@ namespace DocumentService.Controllers
     [ApiController]
     public class DocumentsController : ControllerBase
     {
+        private IAzureBlobService _azureBlobService;
```
Comment (AlexandreSanscartier) [inline DocumentService/Controllers/DocumentsController.cs:17]:
```suggestion
        private IAzureBlobService azureBlobService;
```

File: DocumentService/Controllers/DocumentsController.cs [hunk 2]
```diff
@@ -12,10 +14,23 @@ namespace DocumentService.Controllers
...
 
-        public DocumentsController()
+        public DocumentsController(IAzureBlobService azureBlobService)
         {
+            _azureBlobService = azureBlobService;
```
Comment (AlexandreSanscartier) [inline DocumentService/Controllers/DocumentsController.cs:21]:
```suggestion
            this.azureBlobService = azureBlobService;
```

File: DocumentService/Controllers/DocumentsController.cs [hunk 3]
```diff
@@ -12,10 +14,23 @@ namespace DocumentService.Controllers
...
+        }
+
+        [HttpPost]
+        [Route("v1/documents/test")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public IActionResult UploadDocumentTest(IFormFile file)
+        {
+
+            var result = _azureBlobService.UploadFileAsync(file, AzureBlobContainer.DocumentsContainer).GetAwaiter().GetResult();
```
Comment (AlexandreSanscartier) [inline DocumentService/Controllers/DocumentsController.cs:31]:
```suggestion
            var result = this.azureBlobService.UploadFileAsync(file, AzureBlobContainer.DocumentsContainer).GetAwaiter().GetResult();
```

File: DocumentService/Startup.cs [hunk 4]
```diff
@@ -45,7 +49,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
-                this.setupCodeFirstDevelopmentDatabase();
+                ///this.setupCodeFirstDevelopmentDatabase();
```
Comment (AlexandreSanscartier) [inline DocumentService/Startup.cs:52]:
```suggestion
```

File: DocumentService/Constants/AzureBlobContainer.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DocumentService.Constants
+{
+    public static class AzureBlobContainer
+    {
+
+        public const string DocumentsContainer = "documents";
```
Comment (AlexandreSanscartier) [inline DocumentService/Constants/AzureBlobContainer.cs:11]: Maybe we should move this to the appsettings.json file?

**Review by AlexandreSanscartier — CHANGES_REQUESTED** [review]
A couple of comments to look at.

File: DocumentService/Controllers/DocumentsController.cs (hunk 1 above)
Comment (tc-chris) [reply DocumentService/Controllers/DocumentsController.cs:17]: Done

File: DocumentService/Controllers/DocumentsController.cs (hunk 2 above)
Comment (tc-chris) [reply DocumentService/Controllers/DocumentsController.cs:21]: Done

File: DocumentService/Controllers/DocumentsController.cs (hunk 3 above)
Comment (tc-chris) [reply DocumentService/Controllers/DocumentsController.cs:31]: Do we need "this" here, I feel it's better only in the constructor.

File: DocumentService/Startup.cs (hunk 4 above)
Comment (tc-chris) [reply DocumentService/Startup.cs:52]: This crashes the build

File: DocumentService/Constants/AzureBlobContainer.cs (hunk 5 above)
Comment (tc-chris) [reply DocumentService/Constants/AzureBlobContainer.cs:11]: Done

**Review by AlexandreSanscartier — APPROVED** [review]

**Pull request merged**